MotorControl/src/main/kotlin/org/sagebionetworks/motorcontrol/presentation/TremorStepFragment.kt

```diff
@@ -71,12 +77,29 @@ open class TremorStepFragment: StepFragment() {
         _binding = ComposeQuestionStepFragmentBinding.inflate(layoutInflater, container, false)
         val drawable = step.imageInfo?.loadDrawable(requireContext())
         val tint = step.imageInfo?.tint ?: false
+        val spokenInstructions = SpokenInstructionsConverter.convertSpokenInstructions(
+            step.spokenInstructions,
+            step.duration.toInt(),
+            stepViewModel.nodeState.parent?.node?.hand()?.name
+        )
+        val vibrator = MotorControlVibrator(requireContext())
+        vibrator.vibrate(500)
+        textToSpeech = TextToSpeech(context) {
+            textToSpeech?.speak(spokenInstructions[0], TextToSpeech.QUEUE_ADD, null, "")
+        }
         binding.questionContent.setContent {
             val countdown: MutableState<Long> = remember { mutableStateOf(step.duration.toLong() * 1000) }
             timer = StepTimer(
-                countdown,
-                step.duration,
-                assessmentViewModel::goForward
+                countdown = countdown,
+                stepDuration = step.duration,
+                finished = {
+                    vibrator.vibrate(milliseconds = 500)
+                    // Waiting for speaking to finish before navigating to the next step
+                    while (textToSpeech?.isSpeaking == true) { }
```

Comment:
Usually not a good idea to use a tight loop to wait for something to complete. Would be better to use an [UtterantProgressListener](https://developer.android.com/reference/android/speech/tts/TextToSpeech#setOnUtteranceProgressListener(android.speech.tts.UtteranceProgressListener)) to get notified of completion.